### ISSUE TITLE
Add layer names [ConvMixer]

### ISSUE THREAD

**Issue by IMvision12** [keras-team/keras-cv#1571]

**Comment by Anirban0011** [comment]
@IMvision12 , kindly provide some description on the issue if possible , thanks!

**Comment by IMvision12** [comment]
@Anirban0011 for example below code is of ConvNeXt:

```
def apply_head(x, num_classes, activation="softmax", name=None):
    if name is None:
        name = str(backend.get_uid("head"))
    x = layers.GlobalAveragePooling2D(name=name + "_head_gap")(x)
    x = layers.LayerNormalization(epsilon=1e-6, name=name + "_head_layernorm")(x)
    x = layers.Dense(num_classes, activation=activation, name=name + "_head_dense")(x)
    return x
```

we need to add the names to each layer in the model : `name=name + "_head_gap"` or `"name=name + "_head_layernorm""`

ConvMixer:

def apply_patch_embed(x, dim, patch_size, name=None):
     if name is None:
          name = str(backend.get_uid("patch_embed"))
    x = layers.Conv2D(filters=dim, kernel_size=patch_size, strides=patch_size, name=name + "conv")(x)
    x = tf.nn.gelu(x, name=name + "gelu")
    x = layers.BatchNormalization(name=name + "batchnorm")(x)
    return x

**Comment by Anirban0011** [comment]
Kinda like this ?

`def apply_conv_mixer_layer(x, dim, kernel_size):
    if name is None:
        name = str(backend.get_uid("apply_conv_mixer_layer"))

    residual = x
    x = layers.DepthwiseConv2D(kernel_size=kernel_size, padding="same",name=name+"depthwiseconv" )(x)
    x = tf.nn.gelu(x, name=name + "gelu")
    x = layers.BatchNormalization(name=name + "batchnorm")(x)
    x = layers.Add(name=name + "add")([x, residual])

    x = layers.Conv2D(dim, kernel_size=1,name=name + "conv")(x)
    x = tf.nn.gelu(x,name=name + "gelu")
    x = layers.BatchNormalization(name=name + "batchnorm")(x)
    return x
`

**Comment by Anirban0011** [comment]
Kinda like this ?

```python
def apply_conv_mixer_layer(x, dim, kernel_size):
    if name is None:
        name = str(backend.get_uid("apply_conv_mixer_layer"))

    residual = x
    x = layers.DepthwiseConv2D(kernel_size=kernel_size, padding="same",name=name+"depthwiseconv" )(x)
    x = tf.nn.gelu(x, name=name + "gelu")
    x = layers.BatchNormalization(name=name + "batchnorm")(x)
    x = layers.Add(name=name + "add")([x, residual])

    x = layers.Conv2D(dim, kernel_size=1,name=name + "conv")(x)
    x = tf.nn.gelu(x,name=name + "gelu")
    x = layers.BatchNormalization(name=name + "batchnorm")(x)
    return x
``` 

For layers and functions appearing more than once , should we pass the name parameter at every instance ?

**Comment by IMvision12** [comment]
@Anirban0011 yeah! if there are layers which are repeating then you can just add layer number for eg:
`conv_1`, `conv_2`, also dont forget to add "_" : name=name + "_gelu"

**Comment by Anirban0011** [comment]
> @Anirban0011 yeah! if there are layers that are repeating then you can just add layer numbers for eg: `conv_1`, `conv_2`, also don't forget to add "_" : `name=name + "_gelu"`

Ah!, sure

**Comment by IMvision12** [comment]
Names of layers should be followed by model name: `name=name(model name) + "_avg_pool_0"`, also `apply_conv_mixer_layer` function will have an addition argument `name` which is missing.

Refer this how name argument is passed and used: https://github.com/keras-team/keras-cv/blob/9d012b51ad9342b7fd60078c2c14e782320393b5/keras_cv/models/convnext.py#LL142C2-L142C2

**Comment by Anirban0011** [comment]
Should this be fine ? 

[layer-name-add-convmixer](https://github.com/Anirban0011/keras-cv/blob/layer-name-add-convmixer/keras_cv/models/convmixer.py)

I have not updated the `apply_patch_embed` in case you want to do it.

**Comment by IMvision12** [comment]
Yeah @Anirban0011 do same for apply_patch_embed

**Comment by Anirban0011** [comment]
> Yeah @Anirban0011 do same for apply_patch_embed

Ok , will do if you say so, you want to open a PR for this @IMvision12

**Comment by IMvision12** [comment]
@Anirban0011 you can open PR for this if you want!!! 😃

**Comment by Anirban0011** [comment]
> @Anirban0011 you can open PR for this if you want!!! 😃

Do you want to commit there ?, I would mention this issue either way!

**Comment by Anirban0011** [comment]
[Updated code](https://github.com/Anirban0011/keras-cv/blob/layer-name-add-convmixer/keras_cv/models/convmixer.py)

**Comment by IMvision12** [comment]
https://github.com/Anirban0011/keras-cv/blob/a2a2d56e217e2b22350dc78b7a2085e961909398/keras_cv/models/convmixer.py#L191

UnderScore will be repeated for `apply_conv_mixer_layer` and `apply_patch_embed`

**Comment by Anirban0011** [comment]
> https://github.com/Anirban0011/keras-cv/blob/a2a2d56e217e2b22350dc78b7a2085e961909398/keras_cv/models/convmixer.py#L191
> 
> UnderScore will be repeated for `apply_conv_mixer_layer` and `apply_patch_embed`

Yeah , thought so, check now:

https://github.com/Anirban0011/keras-cv/blob/layer-name-add-convmixer/keras_cv/models/convmixer.py

**Comment by IMvision12** [comment]
Looks Good!

**Comment by Anirban0011** [comment]
> Looks Good!

Should open a PR then ?

**Comment by Anirban0011** [comment]
Got it!
